dev-libs/boost/boost-1.88.0.ebuild

```diff
@@ -207,6 +205,108 @@ multilib_src_compile() {
+	# phoenix: scope/lambda_tests22.cpp(27): test 'x == 1' failed in function 'int main()'
+	# predef: error: Unable to find file or target named (yes, really)
+	# python: AttributeError: property '<unnamed Boost.Python function>' of 'X' object has no setter
+	# qvm: vec_access.hpp:95:223: error: static assertion failed: Boost QVM static assertion failure
+	# regex: regex_timer.cpp:19: ../../../boost/timer.hpp:21:3: error: #error This header is
+	#   deprecated and will be removed. (You can define BOOST_TIMER_ENABLE_DEPRECATED to suppress
+	#   this error.)
+	# serialization: in function `boost::archive::tmpnam(char*)': test_array.cpp:(.text+0x108):
+	#   undefined reference to `boost::filesystem::detail::unique_path(...)'
+	# statechart: TuTestMain.cpp(22) fatal error: in "test_main_caller( argc_ argv )":
+	#   std::runtime_error: Event was not consumed!
+	# unordered: erase_tests.cpp:(.text+0x44cce): undefined reference to
+	#   tbb::detail::r1::execution_slot(tbb::detail::d1::execution_data const*)
+	# wave: t_5_007.cpp(22): error: could not find include file: boost/version.hpp
+	local libs_excluded=(
+		"algorithm" "config" "contract" "geometry" "graph_parallel" "math"
+		"multiprecision" "mysql" "parameter_python" "phoenix" "predef"
+		"python" "qvm" "regex" "serialization" "statechart" "unordered"
+		"wave"
+	)
+
+	if use test-full; then
+		ewarn "Enabling expensive tests - this will take a long time!"
+		libs_excluded=("${libs_excluded[@]/geometry}")
+	fi
+
+	# libraries which must be patched with an additonal main()
+	local libs_needpatch=(
+		"accumulators"
+	)
+
+	# prepare to find libraries without failing candidates
+	local findlibs="find ${BUILD_DIR}/libs -maxdepth 1 -mindepth 1 -type d"
+	for excl in ${libs_excluded[@]}; do
+	   findlibs+=" -not -name ${excl} "
+	done
+
+	# collect libraries to test in sorted order
+	local libs=($(${findlibs} | sort))
+
+	echo -n "Running the following tests: "
+	for lib in ${libs[@]}; do echo -n "$(basename ${lib}) "; done
+	echo
+
+	echo -n "Skipping: "
+	for lib in ${libs_excluded[@]}; do echo -n "${lib} "; done
+	echo
+
+	# Some test suites have no main because normally boost.test can
+	# automatically initialize & run them. Since this only works for
+	# statically linked builds/tests we use an explicit list of tests
+	# which need patching with an additional main().
+	# Determining this dynamically is not really possible.
+	for lib in "${libs_needpatch[@]}"; do
+		# move into library test dir
+		pushd "${BUILD_DIR}/libs/${lib}/test" >/dev/null || die
+			# find all test cases and patch them
+			testcases=($(find . -name "*.cpp"))
+			for testcase in "${testcases[@]}"; do
+				# add main() to bootstrap old-style test suite
+				cat "${FILESDIR}/unit-test-main.cpp" >> ${testcase}
+			done
+		popd >/dev/null
+	done
+
+	# finally build & run all test suites
+	for lib in "${libs[@]}"; do
+		# skip libraries without test directory
+		[[ ! -d "${lib}/test" ]] && continue
```

Comment:
All libraries actually *do* seem to have a test directory, which I didn't know when I initially wrote this. If it turns out to be redundant we can just remove it, though it does not hurt either.

Reply:
A simple sanity check seems to be better than pointlessly dying, so just keep it.